lib/client.js

```diff
@@ -736,37 +737,35 @@ Client.prototype.leaveGroupRequest = function (groupId, memberId) {
 Client.prototype.offsetCommitRequestV2 = function (groupId, memberId, generationId, requests) {
     var self = this;
 
-    return self._waitMetadata().then(function () {
-        return self._findGroupCoordinator(groupId).then(function (connection) {
-            var correlationId = self.nextCorrelationId();
-            var buffer = self.protocol.write().OffsetCommitRequestV2({
-                correlationId: correlationId,
-                clientId: self.options.clientId,
-                groupId: groupId,
-                generationId: generationId,
-                memberId: memberId,
-                retentionTime: self.options.retentionTime,
-                topics: requests
-            }).result;
+    return self._findGroupCoordinator(groupId).then(function (connection) {
+        var correlationId = self.nextCorrelationId();
+        var buffer = self.protocol.write().OffsetCommitRequestV2({
+            correlationId: correlationId,
+            clientId: self.options.clientId,
+            groupId: groupId,
+            generationId: generationId,
+            memberId: memberId,
+            retentionTime: self.options.retentionTime,
+            topics: requests
+        }).result;
 
-            return connection.send(correlationId, buffer).then(function (responseBuffer) {
-                return self.protocol.read(responseBuffer).OffsetCommitResponse().result.topics;
-            });
-        })
-        .then(_mapTopics);
-    });
+        return connection.send(correlationId, buffer).then(function (responseBuffer) {
+            return self.protocol.read(responseBuffer).OffsetCommitResponse().result.topics;
+        });
+    })
+    .then(_mapTopics);
 };
 
 Client.prototype.listGroupsRequest = function () {
     var self = this, buffer, correlationId = self.nextCorrelationId();
 
-    return self._waitMetadata().then(function () {
+    return Promise.resolve().then(function () {
         buffer = self.protocol.write().ListGroupsRequest({
             correlationId: correlationId,
             clientId: self.options.clientId
         }).result;
 
-        return Promise.map(_.values(self.brokerConnections), function (connection) {
+        return Promise.map(_.values(self.initialBrokers), function (connection) {
```

Comment:
@mnorkin Not sure if it can happen that `brokerConnection` won't include all of the `initialBrokers`? I'm asking because `ListGroupsrequest` must be sent to ALL brokers, or you won't receive all groups.

Reply:
How I understood, this command is executed only by groupAdmin and it does not contain any information about the topics, so no topology query here, hence only `initialBrokers` is present. It works if you know the the topology before hand, it does not work in cause you have more brokers, than initially defined.

However I believe this implementation is better than having all the topology read from the all brokers and freeze your application in case you have 100k topics